core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -425,28 +425,74 @@ public void start(boolean isLeader) throws IOException {
       mLostFilesDetectionService = getExecutorService().submit(new HeartbeatThread(
           HeartbeatContext.MASTER_LOST_FILES_DETECTION, new LostFilesDetectionHeartbeatExecutor(),
           Configuration.getInt(PropertyKey.MASTER_HEARTBEAT_INTERVAL_MS)));
-      mStartupConsistencyCheck = getExecutorService().submit(new Callable<List<AlluxioURI>>() {
-        @Override
-        public List<AlluxioURI> call() throws Exception {
-          return checkConsistency(new AlluxioURI("/"), CheckConsistencyOptions.defaults());
-        }
-      });
+      if (Configuration.getBoolean(PropertyKey.MASTER_STARTUP_CONSISTENCY_CHECK_ENABLED)) {
+        mStartupConsistencyCheck = getExecutorService().submit(new Callable<List<AlluxioURI>>() {
+          @Override
+          public List<AlluxioURI> call() throws Exception {
+            return checkConsistency(new AlluxioURI("/"), CheckConsistencyOptions.defaults());
+          }
+        });
+      }
     }
   }
 
   /**
-   * @return the list of inconsistent files at start up, null if the check has not completed or
-   *         failed
+   * Class to represent the status and result of the startup consistency check.
    */
-  public List<AlluxioURI> getStartupConsistencyCheck() {
-    if (mStartupConsistencyCheck.isDone()) {
-      try {
-        return mStartupConsistencyCheck.get();
-      } catch (Exception e) {
-        LOG.warn("Failed to complete start up consistency check.", e);
-      }
+  public static class StartupConsistencyCheck {
```

Comment:
maybe rename to `StartupConsistencyCheckResult`? I was confused for a while thinking that this class actually did the startup consistency check.